analysis.py

```diff
@@ -0,0 +1,716 @@
+        tmp_tickers_file = self.tickers_file.with_suffix(".tmp")
+        success = False
+        # Save to another file
+        try:
+            data_to_save.to_csv(tmp_data_file, index=False)
+            with open(tmp_tickers_file, "w") as f:
+                for ticker in self._tickers:
+                    f.write(ticker + "\n")
+            success = True
+        except Exception as e:
+            print("Unable to save ETFs history")
+            raise e
+
+        # Replace the original files
+        if success:
+            for tmp_file, file in [
+                (tmp_data_file, self.data_file),
+                (tmp_tickers_file, self.tickers_file),
+            ]:
+                try:
+                    os.unlink(file)
+                except FileNotFoundError:
+                    pass
+
+                try:
+                    os.rename(tmp_file, file)
+                except Exception as e:
+                    success = False
+                    raise e
+
+        return success
+
+    def update(self, reload=False):
+        """Fetch the latest price data from server from the last cached date till now.
+
+        Parameters
+        ----------
+        reload
+            If False, will use the stored price cache. Otherwise, will reload all
+            from the server.
+        """
+        today = dt.datetime.now(MOSCOW_TIMEZONE)
+        # If _data have been loaded, only fetch data starting with the last
+        # date in the database
+        if self._data.empty or reload:
+            start_date = EARLIEST_DATE
+        else:
+            start_date = self.last_date - dt.timedelta(days=1)
+
+        if self.verbose:
+            print(f"Updating historical data from {start_date} till {today}")
+
+        new_data, tickers = get_etfs_daily_history(
+            start=start_date, end=today, verbose=self.verbose
+        )
+        if self.verbose:
+            print("Update function received the following new data: ", new_data)
+        if new_data is None or new_data.empty:
+            raise RuntimeError(
+                "The received data frame cannot be empty because one of the "
+                "dates was already present in the history data. "
+                f"Requested dates: {start_date} -- {today}."
+            )
+
+        self._tickers = tickers
+        if self._data.empty or reload:
+            self._data = new_data
+        else:
+            """Drop all data from the original table that repeats in the new
+            one"""
+            old_data = copy.deepcopy(self._data)
+            old_data.drop(
+                old_data[old_data.time >= new_data.time.min()].index, inplace=True
+            )
+
+            merged = old_data.append(new_data, verify_integrity=True, ignore_index=True)
+            # Convert the time column to time because after merge it changes
+            # to object
+            merged.time = pd.to_datetime(self._data.time)
```

Comment:
A bug! @Alexander-Serov #9 

```suggestion
            merged.time = pd.to_datetime(merged.time)
```

Reply:
Вот, это мощно, согласен!